dist/src/test/java/io/camunda/application/commons/backup/BackupPrioritiesTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.backup;
+
+import static com.tngtech.archunit.core.domain.JavaClass.Predicates.implement;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import io.camunda.operate.property.OperateProperties;
+import io.camunda.tasklist.property.TasklistProperties;
+import io.camunda.webapps.schema.descriptors.backup.BackupPriority;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class BackupPrioritiesTest {
+
+  static final List<JavaClass> ALL_IMPLEMENTATIONS;
+
+  static {
+    ALL_IMPLEMENTATIONS =
+        new ClassFileImporter()
+                .importPackages(
+                    // TODO ADD optimize
+                    "io.camunda.webapps.schema") // , "io.camunda.optimize.service.db.schema.index")
+                .that(implement(BackupPriority.class))
+                .stream()
+                .filter(clz -> !clz.getName().contains("Abstract"))
```

Comment:
❓️ Is there no helper in ArchUnit to filter out abstract classes, and not rely on naming conventions? Seems quite brittle.

Reply:
So for example all optimize indices follow this pattern (different from the other webapps):
```java
public abstract class BusinessKeyIndex<TBuilder> extends DefaultIndexMappingCreator<TBuilder>
    implements Prio6Backup
``
This is the class we actually want to test against, but unfortunately it's abstract. 
The concrete implementations are  `BusinessKeyIndexES` and  `BusinessKeyIndexOS`.